src/dune_rules/super_context.ml

```diff
@@ -393,13 +393,19 @@ let create ~(context : Context.t) ~host ~packages ~stanzas =
   let* root_expander =
     let* artifacts_host, context_host =
       match context.for_host with
-      | None -> Memo.return (artifacts, context)
+      | None -> Memo.return (artifacts, None)
       | Some host ->
         let+ artifacts = Artifacts_db.get host in
-        (artifacts, host)
+        (artifacts, Some host)
     in
     let* scope = Scope.DB.find_by_dir context.build_dir in
-    let* scope_host = Scope.DB.find_by_dir context_host.build_dir in
+    let* scope_host =
+      match context_host with
+      | None -> Memo.return None
+      | Some context_host ->
+        let+ scope = Scope.DB.find_by_dir context_host.build_dir in
+        Some scope
+    in
```

Comment:
There's another piece of code in this file that looks suspicious:

```
  let expander_for_artifacts ~scope ~external_env ~root_expander ~dir =
    Expander.extend_env root_expander ~env:external_env
    |> Expander.set_scope ~scope |> Expander.set_dir ~dir
```

It doesn't seem right to set the scope without setting the scope_host appropriately. I guess we can fix this on another occasion.

Reply:
I was just typing this in another comment.

Reply:
I think the current fix is in the wrong place.

Reply:
There's a simpler way to derive the scope_host from the scope that perhaps we should follow:

1. Get the root directory of `scope`.
2. Transform this directory to the analogous directory in the host's context
3. Call `Scope.DB.find_by_dir` on it.

This kind of complicates my refactoring of making source tree context based, but it's ok for now.

Reply:
So for this change, you would need a `; context_host : Context.t option` field rather than a `scope_host`.

Reply:
Or perhaps we should modify `set_scope` to take both the `scope` and the `scope_host`.  and move this logic to whoever is calling `set_scope.` It seems good to keep Expander relatively oblivious to cross compilation logic.

Reply:
I pushed the version where `scope_host` is derived from context. I'll explore the version where `set_scope` takes `scope_host` now and see if I like it better.

Reply:
Pushed the `set_scope` fix instead, which is a lot more minimal. Let me know what you think.